Customer view

sebastianbot6969 opened the pull request.

sebastianbot6969 committed:
Merge branch 'main' of https://github.com/P3-DAT3-03/Flea into CustomerView

sebastianbot6969 committed:
Merge remote-tracking branch 'origin/Modals' into CustomerView

sebastianbot6969 committed:
A not not working modal to make a new customer

- The function does not work yet

sebastianbot6969 committed:
The Modal for both create and edit customer works

- In CustomerOverview.razor the create customer modal is working, but with no css implemented
 - As well as some changes on the dummy data is written, so it is towards the data from the database.
- In Modal.razor changes has been made, to make the modal run smooth without closing and making it easier to lock css to a modal in the future.
- In UserListItem.razor there has been made the modal for edit a customer.
 TL;DR
small changes

sebastianbot6969 committed:
The Modal for both create and edit customer works without CSS

- In CustomerOverview.razor the create customer modal is working with the CSS in progress
 - A validation component in works for information on customer.
- In UserListItem.razor the modal for edit a customer is working, but with the CSS in progress.
 - A validation component in works for information on customer.
- In Customer.cs a datatype has been implemented to control it only can be digits.
- A New file CustomerOverview.razor.css is a CSS file for the modal in creating or editing a customer.

 TL;DR
small changes

sebastianbot6969 committed:
Merge branch 'main' of https://github.com/P3-DAT3-03/Flea into CustomerView

 Conflicts:
	Flea/Models/Customer.cs
	Flea/Pages/CustomerOverview.razor
	Flea/Shared/Modal.razor

sebastianbot6969 committed:
The customer overview is now finished with working functions.
- The file UserListItem.razor has been renamed as CustomerUserListItem.razor, and with add css fil CustomerUserListItem.razor.css.
- CustomerForm.razor is the form made for the modal to create and edit a customer, with added css file to it CustomerForm.razor.css
- Both CustomerOverview.razor and CustomerUserListItem.razor now has implemented the CustomerForm.razor for the modal popup.
- Customer.cs do now have an function to clone, this is to create a customer.

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -1,53 +1,53 @@
 @page "/CustomerOverview"
-@using System.Data.Common
+@using Customer = Flea.Models.Customer
 <div>
```
Husk at smide en enter så der er linje afstand. Formatering :)

jonasKjellerup commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -1,53 +1,53 @@
-    private List<Models.Customer> _customers = new()
+    // Dummy data ---
+
+    private List<Customer> Customers = new()
```
private field `Customers` does not adhere to naming conventions

jonasKjellerup commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -1,53 +1,53 @@
     // ---
 
-    private IEnumerable<Models.Customer> _shownCustomers;
+    private IEnumerable<Customer> ShownCustomers;
```
private field `ShownCustomers` does not adhere to naming conventions

pimber replied:
_showCustomer

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -1,53 +1,53 @@
     </div>
-
-    @foreach (var customer in _shownCustomers)
+    @foreach (var customer in ShownCustomers)
```
_showCustomers

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -1,53 +1,53 @@
     {
-        _customers = _customers.Where(c => c.Id != id).ToList();
-        _shownCustomers = _shownCustomers.Where(c => c.Id != id);
+        Customers = Customers.Where(c => c.Id != id).ToList();
+        ShownCustomers = ShownCustomers.Where(c => c.Id != id);
```
Kig på name convention

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -57,31 +57,31 @@
         if (input == "")
         {
-            _shownCustomers = _customers;
+            ShownCustomers = Customers;
```
nameing

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -57,31 +57,31 @@
+
             var results = from c
-                in _customers
+                in Customers
```
naming

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -57,31 +57,31 @@
     // Finds by name
             var results = from c
-                in _customers
+                in Customers
```
naming

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -57,31 +57,31 @@
                 where c.Name.ToLower().Contains(input.ToLower())
                 select c;
-            _shownCustomers = results;
+            ShownCustomers = results;
```
naming

pimber commented on Flea/Pages/CustomerOverview.razor:
```diff
@@ -57,31 +57,31 @@
     protected override void OnInitialized()
     {
-        _shownCustomers = _customers;
+        ShownCustomers = Customers;
```
nameing

pimber commented on Flea/Shared/Modal.razor:
```diff
@@ -1,9 +1,9 @@
 @if (_showModal)
 {
-    <div class="modal fade show" id="myModal" style="display:block; background-color: transparent;" aria-modal="true" role="dialog">
+    <div class="modal fade show" id="myModal" style="display: block; background-color: transparent" aria-modal="true" role="dialog">
```
Inline style skal laves som en class og smides på class="modal fade show <et eller andet navn>"

pimber commented on Flea/Shared/Modal.razor:
```diff
@@ -1,9 +1,9 @@
+    <div class="modal fade show" id="myModal" style="display: block; background-color: transparent" aria-modal="true" role="dialog">
         <div class="modal-dialog">
-            <div class="modal-content">
+            <div class="modal-content" >
```
Fjern mellemrum

pimber commented on Flea/Shared/Modal.razor:
```diff
@@ -16,8 +16,8 @@
+
     private bool _showModal = false;
-    private void ModalShow() => _showModal = true;
+    private void ModalShow()=> _showModal = true;
```
mellemrum

jonasKjellerup commented on Flea/Shared/CustomerUserListItem.razor:
```diff
@@ -0,0 +1,50 @@
+</div>
+
+@code {
+    Modal _child;
```
should be marked as `private`

sebastianbot6969 committed:
Naming convention fixed along

- CustomerOverview.razor fixed the naming after the C# standard
- Modal.razor deleted the styling and making a class in css for the style
- Modal.razor.css Made a class for the style in Modal.razor

pimber commented on Flea/Shared/CustomerForm.razor:
```diff
@@ -0,0 +1,66 @@
+        @*     <InputText @bind-Value="Temp.Comments"></InputText> *@
+        @* </div> *@
+        <div id="move-down">
+            <button type="submit" class="button create-button button--outline">@(Customer.Id == 0 ? "Opret Kunde" : "Gem ændringer")</button>
```
Opret loppe

sebastianbot6969 committed:
Opret kunde -> Opret loppe

Small changes
